backend/de.metas.acct.base/src/main/sql/postgresql/ddl/functions/BusinessPartnerAccountSheetReport.sql

```diff
@@ -127,42 +130,44 @@ BEGIN
                                                 rowid,
                                                 ad_org_id,
                                                 targetCurrencyCode,
-                                                doctype)
-	SELECT--
-		  i.beginningBalance,
-		  i.amount,
-		  i.endingBalance,
-		  i.dateacct,
-		  i.description,
-		  i.c_doctype_id,
-		  i.documentno,
-		  i.created,
-		  i.c_currency_id,
-		  row_number() OVER (),
-		  i.ad_org_id,
-		  (SELECT iso_code
-		   FROM c_currency c
-					INNER JOIN c_acctschema accts ON c.c_currency_id = accts.c_currency_id
-					INNER JOIN ad_clientinfo ac ON accts.c_acctschema_id = ac.c_acctschema1_id
-		   LIMIT 1)      targetCurrencyCode,
-		  (CASE
-			   when i.c_doctype_id < 0 THEN
-				   (select coalesce(rt.name, r.name) as doctype
-					from ad_ref_list r
-							 join ad_ref_list_trl rt
-								  on r.ad_ref_list_id = rt.ad_ref_list_id and rt.ad_language = p_ad_language
-					where ad_reference_id = 183
-					  and value = 'CMA'
-				   )
-			   ELSE
-				   (SELECT dtt.name
-					FROM c_doctype dt
-							 INNER JOIN c_doctype_trl dtt ON dt.c_doctype_id = dtt.c_doctype_id
-					WHERE dtt.ad_language = p_ad_language
-					  AND i.c_doctype_id = dt.c_doctype_id)
-			  END
-			  )::text AS docType
-	FROM invoicesAndPaymentsInPeriod i;
+                                                doctype,
+                                                issotrx)
```

Comment:
line 147:
> SELECT iso_code

it shall be 
> SELECT c.iso_code

IMPORTANT: never use a field without alias when JOINs are involved.

cc @cristinamghita